src/main/java/io/singularitynet/MissionHandlers/VideoProducerImplementation.java

```diff
@@ -54,27 +54,54 @@ public int getWidth()
     @Override
     public int getHeight() { return this.videoParams.getHeight(); }
 
-    private ByteBuffer getRGBFrame()
+    private AbstractMap.Entry<ByteBuffer, int[]> getRGBFrame()
     {
         Framebuffer framebuffer = MinecraftClient.getInstance().getFramebuffer();
         int i = framebuffer.textureWidth;
         int j = framebuffer.textureHeight;
+        long time1 = System.nanoTime();
         ImageClass imageclass = new ImageClass(i, j, false);
-        RenderSystem.bindTexture(framebuffer.getColorAttachment());
-        imageclass.loadFromTextureImage(0, true);
-        imageclass.mirrorVertically();
-        byte[] ni_bytes;
-        try {
-             ni_bytes = imageclass.getBytes();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-        imageclass.close();
-        if (ni_bytes.length <= 0)
-            return null;
-        else {
-            return ByteBuffer.wrap(ni_bytes);
-        }
+        long time2 = System.nanoTime();
```

Comment:
We can save performance stats using 
Profiler profiler = MinecraftClient.getInstance().world.getProfiler();

then you start profiling with:
profiler.push("getRGBFrame");

Reply:
and stop with profiler.pop("getRGBFrame");